cleverhans/experimental/certification/utils.py

```diff
@@ -244,3 +246,82 @@ def minimum_eigen_vector(x, num_steps, learning_rate, vector_prod_fn):
   for _ in range(num_steps):
     x = eig_one_step(x, learning_rate, vector_prod_fn)
   return x
+
+
+def tf_lanczos_smallest_eigval(vector_prod_fn,
+                               matrix_dim,
+                               max_iter=1000,
+                               collapse_tol=1e-12,
+                               dtype=tf.float32):
+  """Computes smallest eigenvector and eigenvalue using Lanczos in pure TF.
+
+  Args:
+    vector_prod_fn: function which takes a vector as an input and returns
+      matrix vector product.
```

Comment:
Can you write this with some high level math notation or point to https://en.wikipedia.org/wiki/Lanczos_algorithm#The_algorithm so people know what you are talking about